Assets/Content/Furniture/Machines/Science/CellCharger.cs

```diff
@@ -49,10 +54,58 @@ private void Recharge()
             IChargeable chargeable = item.GetComponent<IChargeable>();
             if (chargeable != null)
             {
-		// Effectively add charge
                 chargeable.AddCharge(chargeable.GetChargeRate());
+                UpdateLights(chargeable.GetPowerPercentage());
             }
-            
+        }
+    }
+    
+    /// <summary>
+    /// Turn off all lights
+    /// </summary>
+    private void TurnOffLights() 
+    {
+        redLight.TurnLightOff();
+        orangeLight.TurnLightOff();
+        yellowLight.TurnLightOff();
+        greenLight.TurnLightOff();
+    }
+
+    /// <summary>
+    /// Turns on lights based on powerPercentage.
+    /// Power groups are (0 - .25), (.25 - .50), (.50 - .75), (.75 - 1) 
+    /// </summary>
+    /// <param name="powerPercentage">Value between 0 - 1</param>
+    private void UpdateLights(float powerPercentage) 
+    {
+        if (0 <= powerPercentage && powerPercentage < .25f) // power @ 25%
+        {
+            redLight.MakeLightBlink();
+        }
+        else if (.25f <= powerPercentage && powerPercentage < .50f) // power @ 50%
+        {
+            redLight.MakeLightStayOn();
+            orangeLight.MakeLightBlink();
+        }
+        else if (.50f <= powerPercentage && powerPercentage < .75f) // power @ 75%
+        {
+            redLight.MakeLightStayOn();
+            orangeLight.MakeLightStayOn();
+            yellowLight.MakeLightBlink();
+        }
+        else if  (.75f <= powerPercentage && powerPercentage < 1f)
+        {
+            redLight.MakeLightStayOn();
+            orangeLight.MakeLightStayOn();
+            yellowLight.MakeLightStayOn();
+            greenLight.MakeLightBlink();
+        }
+        else if (1f <= powerPercentage)
+        {
+            redLight.MakeLightStayOn();
+            orangeLight.MakeLightStayOn();
+            yellowLight.MakeLightStayOn();
+            greenLight.MakeLightStayOn();
         }
```

Comment:
By using the lights array, this could be simplified by taking the increment (`1f / lights.Length`) and then calculating the index of the blinking light. This tells you which lights should be on (those before the blinking light) and off too (those after the blinking light).

Reply:
I got this working locally, but couldn't figure out to incorporate it with the networking changes from @Ryan089. The changes are pushed up to the PR for anyone wanting to rewrite  the cell charger to use an array of lights.

Reply:
Recommendation supported, but left for consideration for future work.